src/writer/autogen.py

```
@@ -253,19 +368,227 @@ def _get_main_prompt(description: str):
+    if error_messages:
+        autogen_state["messages"].extend(error_messages)
+        return None
+
+    if not links:
+        return autogen_state["final_graph"]
+
+    autogen_state["actions"].append({"type": "link", "links": links})
+    return id_to_node
+
+
+def _call_tools(tool_calls: Mapping[str, Mapping], autogen_state: AutogenState):
```

Pylint (3.3.7)

[refactor] Either all return statements in a function should return an expression, or none of them should.

(R1710)